File: app/src/main/java/org/xdty/callerinfo/di/modules/PhoneStatusModule.java
package org.xdty.callerinfo.di.modules;

import org.xdty.callerinfo.contract.PhoneStateContract;
import org.xdty.callerinfo.presenter.PhoneStatePresenter;

import dagger.Module;
import dagger.Provides;

@Module
public class PhoneStatusModule {

    PhoneStateContract.View mView;

    public PhoneStatusModule(PhoneStateContract.View view) {
        mView = view;
    }

    @Provides
    PhoneStateContract.View provideView() {
        return mView;
    }

    @Provides
    PhoneStateContract.Presenter providePresenter() {
        return new PhoneStatePresenter(mView);
    }
}

File: app/src/main/java/org/xdty/callerinfo/di/modules/PhoneStatusModule.kt
package org.xdty.callerinfo.di.modules

import dagger.Module
import dagger.Provides
import org.xdty.callerinfo.contract.PhoneStateContract.Presenter
import org.xdty.callerinfo.contract.PhoneStateContract.View
import org.xdty.callerinfo.presenter.PhoneStatePresenter
import javax.inject.Singleton

@Module
class PhoneStatusModule(private var mView: View) {
    @Provides
    internal fun provideView(): View {
        return mView
    }

    @Singleton
    @Provides
    internal fun providePresenter(): Presenter {
        return PhoneStatePresenter(mView)
    }

}